src/adapter/src/coord/sequencer/inner/cluster.rs

```diff
@@ -217,18 +233,45 @@ impl Coordinator {
 
         let new_workload_class = new_config.workload_class.clone();
         match (&config.variant, &new_config.variant) {
-            (Managed(_), Managed(_)) => {
-                self.sequence_alter_cluster_managed_to_managed(
-                    Some(session),
-                    cluster_id,
-                    new_config,
-                    ReplicaCreateDropReason::Manual,
-                )
-                .await?;
+            (Managed(_), Managed(new_config_managed)) => {
+                let alter_followup = self
+                    .sequence_alter_cluster_managed_to_managed(
+                        Some(session),
+                        cluster_id,
+                        new_config.clone(),
+                        ReplicaCreateDropReason::Manual,
+                        strategy.clone(),
+                    )
+                    .await?;
+                return match alter_followup {
+                    FinalizationNeeded::In(duration) => {
+                        let span = Span::current();
+                        let new_config_managed = new_config_managed.clone();
+                        Ok(StageResult::Handle(mz_ore::task::spawn(
+                            || "Finalize Alter Cluster",
+                            async move {
+                                tokio::time::sleep(duration).await;
+                                let stage = ClusterStage::Finalize(FinalizeAlterCluster {
+                                    validity,
+                                    plan,
+                                    new_config: new_config_managed,
+                                });
+                                Ok(Box::new(stage))
```

Comment:
Any specific reason for boxing here?

Reply:
The stage trait defines a function `stage` that returns ` Result<StageResult<Box<Self>>, crate::AdapterError>`, Doing the boxed return here lets me return directly from this function in the `stage` fn for `ClusterStage`